test_adjoint/test_demos.py

```diff
@@ -4,27 +4,70 @@
 
 import glob
 import os
+import re
 import shutil
-import subprocess
-import sys
 from os.path import splitext
 
 import pytest
 
-# Set environment variable to specify that tests are being run so that we can
-# cut down the length of the demos
-os.environ["GOALIE_REGRESSION_TEST"] = "1"
+from goalie.log import *
 
 cwd = os.path.abspath(os.path.dirname(__file__))
 demo_dir = os.path.abspath(os.path.join(cwd, "..", "demos"))
 all_demos = glob.glob(os.path.join(demo_dir, "*.py"))
 
+# Modifications dictionary to cut down run time of demos:
+# - Each key is the name of a demo script to be modified
+# - Each value is a dictionary where:
+# -- The key is the original string or regex pattern that we wish to replace
+# -- The value is the replacement string (use "" to remove the original code)
+modifications = {
+    "burgers-hessian.py": {""""maxiter": 35""": """"maxiter": 3"""},
+    "gray_scott.py": {
+        "end_time = 2000.0": "end_time = 10.0",
+        r"ic = mesh_seq.get_initial_condition\(\)\n.*?adj_outfile.write\(\*adj_end.subfunctions\)": "",
```

Comment:
These bits are kind of ugly but hopefully we can clean them up when we address #117.

Reply:
Indeed, also hard to immediately see what's happening. I'll keep an eye on it and see if I can improve it :)